### PR TITLE
Add validation and transactional support

### PR DESCRIPTION
**Pull request by chertiav**
- added a CustomGlobalExceptionHandler class;
- added meaningful validation for CreateBookRequestDto;
- added @Transactional annotations to BookServiceImpl methods.

### CODE REVIEW

File: src/main/java/com/achdev/onlinebookstoreapp/dto/CreateBookRequestDto.java [hunk 1]
```diff
@@ -1,16 +1,32 @@
...
     private String title;
+
+    @NotBlank(message = "Author is mandatory")
     private String author;
+
+    @Pattern(regexp = "^(97(8|9))?\\d{9}(\\d|X)$",
+            message = "Invalid ISBN format")
     private String isbn;
+
+    @Digits(integer = 6,
```
Comment (Elena-Bruyako) [inline src/main/java/com/achdev/onlinebookstoreapp/dto/CreateBookRequestDto.java:27]: Check if null value could be stored here

File: src/main/java/com/achdev/onlinebookstoreapp/exception/CustomGlobalExceptionHandler.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+            @NonNull HttpStatusCode status,
+            @NonNull WebRequest request) {
+        Map<String, Object> errors = new LinkedHashMap<>();
+        errors.put("timestamp", LocalDateTime.now());
+        errors.put("status", HttpStatus.BAD_REQUEST);
+        List<String> errorMessages = ex.getBindingResult().getAllErrors().stream()
+                .map(this::getErrorMessage)
+                .toList();
+        errors.put("errors", errorMessages);
+        return new ResponseEntity<>(errors, headers, status);
```
Comment (Elena-Bruyako) [inline src/main/java/com/achdev/onlinebookstoreapp/exception/CustomGlobalExceptionHandler.java]:
```suggestion
        return new ResponseEntity<>(errors, headers, HttpStatus.BAD_REQUEST);
```

File: src/main/java/com/achdev/onlinebookstoreapp/dto/CreateBookRequestDto.java [hunk 3]
```diff
@@ -1,16 +1,32 @@
...
 @Data
 @JsonIgnoreProperties(ignoreUnknown = true)
 public class CreateBookRequestDto {
+    @NotBlank(message = "Title is mandatory")
     private String title;
+
+    @NotBlank(message = "Author is mandatory")
     private String author;
+
+    @Pattern(regexp = "^(97(8|9))?\\d{9}(\\d|X)$",
```
Comment (Elena-Bruyako) [inline src/main/java/com/achdev/onlinebookstoreapp/dto/CreateBookRequestDto.java:22]: `@NotBlank` ?

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comments

File: src/main/java/com/achdev/onlinebookstoreapp/dto/CreateBookRequestDto.java (hunk 1 above)
Comment (chertiav) [reply src/main/java/com/achdev/onlinebookstoreapp/dto/CreateBookRequestDto.java:27]: ++

File: src/main/java/com/achdev/onlinebookstoreapp/exception/CustomGlobalExceptionHandler.java (hunk 2 above)
Comment (chertiav) [reply src/main/java/com/achdev/onlinebookstoreapp/exception/CustomGlobalExceptionHandler.java]: ++

File: src/main/java/com/achdev/onlinebookstoreapp/dto/CreateBookRequestDto.java (hunk 3 above)
Comment (chertiav) [reply src/main/java/com/achdev/onlinebookstoreapp/dto/CreateBookRequestDto.java:22]: ++

**Review by Elena-Bruyako — APPROVED** [review]

**Pull request merged**